src/LinkedListVisualizer/LinkedListVisualizer.jsx

```diff
@@ -0,0 +1,497 @@
+                            onChange={this.handleCycleBox}
+                            >
+                        </input>
+                    </div>
+
+
+                <button
+                    type="button"
+                    onClick={this.handleAddCycle}
+                    >
+                    Create Cycle
+                </button>
+            </form>
+
+            <br></br>
+
+            <button onClick={() => this.visualizeTransversal()}>
+                      Visualize
+            </button>
+
+            <br></br>
+            <br></br>
+
+
+            <button
+                    type="button"
+                    onClick={this.handleClearBoard}
+                    >
+                    Clear Board
+            </button>
+           
+            </center>
+
+            <div className="pushLeft">
+            <div className="flexRow">
+                {grid.map((node) => {
+                    const {location, data, nextNode} = node;
+                    
+                    if(location === (numberOfNodes - 1) && this.state.cyclePresent){
+                        return(
+                            <div>
+
+                            <div className="flexRow">
+                            <Node 
+                            location={location}
+                            data={data}
+                            key={location}
+                            nextNode={nextNode}
+                            />
+                            </div>
+                            
+                            <div style={{height: 100, backgroundColor: "#435058", width: this.state.downArrowWidth, position: "relative", left: (this.state.nodeWidth - this.state.downArrowWidth) / 2 }}>
+                            </div>
+
+                            </div>
+                        )
+                    } else if(location == this.state.cycleTo && this.state.cyclePresent){
+                        return(
+                            <div>
+
+                            <div className="flexRow">
+                            <Node 
+                            location={location}
+                            data={data}
+                            key={location}
+                            nextNode={nextNode}
+                            />
+                            <Pointer></Pointer>
+
+                            </div>
+                            <div style={{position: "relative", left: 10}}>
+                                <div className="upTriangle">
+                                </div>
+                            </div>
+
+                            <div style={{height: 100 - this.state.downArrowWidth, backgroundColor: "#435058", width: this.state.downArrowWidth, position: "relative", left: (this.state.nodeWidth - this.state.downArrowWidth) / 2 }}>
+                            </div>
+                            </div>
+                            )} else{
```

Comment:
There's a lot of common code in these if-else branches, I would suggest using some variables to reduce the duplication. (remember you can store components/`<div>`s in a variable)